hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoUpdateTransitionLogic.java

```diff
@@ -70,120 +62,79 @@
+		final var accountId = Id.fromGrpcAccount(op.getAccountIDToUpdate());
+		final var account = accountStore.loadPossiblyDetachedAccount(accountId);
 
-	private ResponseCodeEnum sanityCheck(AccountID target, HederaAccountCustomizer customizer) {
-		if (!ledger.exists(target) || ledger.isSmartContract(target)) {
-			return INVALID_ACCOUNT_ID;
+		if (account.isDetached()) {
+			validateTrue(affectsExpiryAtMost(op), ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
 		}
-
-		final var changes = customizer.getChanges();
-		final var keyChanges = customizer.getChanges().keySet();
-
-		if (ledger.isDetached(target)) {
-			if (!keyChanges.equals(EXPIRY_ONLY)) {
-				return ACCOUNT_EXPIRED_AND_PENDING_REMOVAL;
-			}
-		}
-
-		if (keyChanges.contains(AccountProperty.EXPIRY)) {
-			final long newExpiry = (long)changes.get(AccountProperty.EXPIRY);
-			if (newExpiry < ledger.expiry(target)) {
-				return EXPIRATION_REDUCTION_NOT_ALLOWED;
-			}
+		if (op.hasExpirationTime()) {
+			validateTrue(validator.isValidExpiry(op.getExpirationTime()), INVALID_EXPIRATION_TIME);
+			validateTrue(op.getExpirationTime().getSeconds() > account.getExpiry(), EXPIRATION_REDUCTION_NOT_ALLOWED);
 		}
-
-		if (keyChanges.contains(AccountProperty.MAX_AUTOMATIC_ASSOCIATIONS)) {
-			final long newMax = (int) changes.get(AccountProperty.MAX_AUTOMATIC_ASSOCIATIONS);
-			if (newMax < ledger.alreadyUsedAutomaticAssociations(target)) {
-				return EXISTING_AUTOMATIC_ASSOCIATIONS_EXCEED_GIVEN_LIMIT;
-			}
-			if (newMax > dynamicProperties.maxTokensPerAccount()) {
-				return REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT;
-			}
+		if (op.hasMaxAutomaticTokenAssociations()) {
+			validateTrue(op.getMaxAutomaticTokenAssociations().getValue() >= account.getAlreadyUsedAutomaticAssociations(), EXISTING_AUTOMATIC_ASSOCIATIONS_EXCEED_GIVEN_LIMIT);
+			validateTrue(op.getMaxAutomaticTokenAssociations().getValue() <= dynamicProperties.maxTokensPerAccount(), REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT);
 		}
 
-		return OK;
+		/* --- Do the business logic --- */
+		account.updateFromGrpc(
+				optional(op.hasKey(), op.getKey()),
+				optional(op.hasMemo(), op.getMemo()),
+				optional(op.hasAutoRenewPeriod(), op.getAutoRenewPeriod()),
+				optional(op.hasExpirationTime(), op.getExpirationTime()),
+				optional(op.hasProxyAccountID(), op.getProxyAccountID()),
+				optional(op.hasReceiverSigRequiredWrapper(), op.getReceiverSigRequiredWrapper()),
+				optional(op.hasMaxAutomaticTokenAssociations(), op.getMaxAutomaticTokenAssociations())
+		);
+
+		/* --- Persist the models --- */
+		accountStore.persistAccount(account);
+		/* --- Externalise the transaction effects in the record stream */
+		transactionRecordService.includeChangesToAccount(account);
 	}
 
-	private HederaAccountCustomizer asCustomizer(CryptoUpdateTransactionBody op) {
-		HederaAccountCustomizer customizer = new HederaAccountCustomizer();
+	private Optional optional(boolean has, Object value) {
```

Comment:
nit: If the author prefers to keep the helper - I think that the first argument could be named as "is present" to be consistent with the `Optional.class` - related semantics.